runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargesOverlay.java

```diff
@@ -61,41 +64,36 @@ public Dimension render(Graphics2D graphics)
 	{
 		graphics.setFont(FontManager.getRunescapeSmallFont());
 
-		for (WidgetItem item : getJewelleryWidgetItems())
+		for (WidgetItem widgetItem : getWidgetItems())
 		{
-			JewelleryCharges charges = JewelleryCharges.getCharges(item.getId());
+			final ItemComposition item = client.getItemDefinition(widgetItem.getId());
+			final Integer charges = ChargeCounts.getCharges(item, config);
 
 			if (charges == null)
 			{
 				continue;
 			}
 
-			final Rectangle bounds = item.getCanvasBounds();
-			final TextComponent textComponent = new TextComponent();
-			textComponent.setPosition(new Point(bounds.x, bounds.y + 16));
-			textComponent.setText(String.valueOf(charges.getCharges()));
-			textComponent.render(graphics);
+			final Rectangle bounds = widgetItem.getCanvasBounds();
+			final int x = bounds.x, y = bounds.y + 16;
+
+			graphics.setColor(Color.BLACK); // Text shadow
```

Comment:
Even though this increases performance by a small bit, this should be changed back to text component as using text component makes it easier to make the UI look consistent and easily changeable in future.